src/core/config.rs

```diff
@@ -91,12 +96,22 @@ impl KeybindingConfig {
             "delete" => Some(Event::Key(Key::Delete)),
             "insert" => Some(Event::Key(Key::Insert)),
             "escape" => Some(Event::Key(Key::Esc)),
+            "ctrl+right" => Some(Event::Unsupported(vec![27, 91, 49, 59, 53, 67])),
```

Comment:
I'm a bit confused: why are these named "ctrl+something" instead of an actual command name like "word-left" or "word-right"? Other question: what are `ctrl+shift+up` and `ctrl+shift+down` supposed to do? It seems that it's not doing anything visible at the moment.

Reply:
> I'm a bit confused: why are these named "ctrl+something" instead of an actual command name like "word-left" or "word-right"?

It looks different on HEAD. There its separated. a bit clearer.

But in principle there are two parsers: One is to parse a command name to `Command`-struct (e.g. `select_all`). 

The other is to parse key-descriptions to `Termion::Event`, e.g. `escape` to `Key::Esc`. 
Now a user can edit their JSON file to let pressing `escape` trigger `select_all` for example.

`ctrl+right` and others are not supported by Termion right now, as they differ from Terminal to Terminal, which...sucks. But to have the functionality, I basically used the Unsupported event to make it work with *my* terminal. 
So this is already a "known bug" of this implementation. Another is, that Termion doesn't differentiate between `ctrl+a` and `ctrl+shift+a`. I have opened a bugreport there, but I fear its not really actively maintained anymore. 

> Other question: what are ctrl+shift+up and ctrl+shift+down supposed to do? It seems that it's not doing anything visible at the moment.

Try typing after doing that :-) It adds a cursor below or above the current cursor, which are unfortunately invisible at the moment. I have some screencasts showing of the features, just have to find a place to upload to

Reply:
Ah, sorry. I confused `ctrl+shift+up/down` with `alt+shift+up/down`.
`ctrl+shift+up/down` is supposed to move the current line one line up or down (swap lines). But this is currently not supported by xi-core ( [#793](https://github.com/xi-editor/xi-editor/issues/793) ).

Either way, this part is *not* there to define functionality! Its there so that users can use `ctrl+shift+up/down` for whatever they want (They could map the function of  `alt+shift+up/down` to `ctrl+shift+up/down`. Or they could for whatever reason map `ctrl+shift+up` to `select_all` and `ctrl+shift+down` to `next buffer`.).
This part of the code is only there to "know a key-combination exists". Not connect it to any functionality.

But for completeness, here are the screencasts I talked about (just "view raw" for each file):
https://gist.github.com/msirringhaus/83fbc1ef10ae93e56fe8d68dea1dd3b3

"switchcase" is what  `alt+shift+up/down` is doing at the moment.
`ctrlD` obviously is what `ctrl+d` does.
And `Palette` is the CommandPrompt with suggestions. There is a crate to have sublime-like fuzzy-search, which I want to implement next. Then finally, the Prompt should handle up/down movements to select suggestions (at the moment it only shows them)